src/main/java/org/ebml/matroska/MatroskaFileWriter.java

```diff
@@ -35,37 +38,84 @@
 
   protected DataWriter ioDW;
 
-  private final MatroskaFileMetaSeek metaSeek;
-  private final MatroskaFileCues cueData;
-  private final MatroskaCluster cluster;
-  private final MatroskaSegmentInfo segmentInfoElem;
-  private final MatroskaFileTracks tracks;
-  private final MatroskaFileTags tags;
+  private MatroskaFileMetaSeek metaSeek;
+  private MatroskaFileCues cueData;
+  private MatroskaCluster cluster;
+  private MatroskaSegmentInfo segmentInfoElem;
+  private MatroskaFileTracks tracks;
+  private MatroskaFileTags tags;
+
+  private Long defferedTimecodeScale;
+  private Double defferedDuration;
+  private List<MatroskaFileTrack> defferedTracks;
+  private List<MatroskaFileTagEntry> defferedTags;
+  private boolean initialized = false;
 
   /**
    * @param outputDataWriter DataWriter to write out to.
    */
   public MatroskaFileWriter(final DataWriter outputDataWriter)
   {
     ioDW = outputDataWriter;
+  }
+
+  void initialize() {
```

Comment:
For consistency with the style of the existing code, the opening brace should be on its own line. This applies to several other places below.